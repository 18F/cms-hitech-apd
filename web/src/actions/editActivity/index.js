export * from './contractorResources';
export * from './costAllocate';
export * from './overview';
