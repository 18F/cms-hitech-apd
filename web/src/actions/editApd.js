import { updateBudget } from './apd';

export const ADD_APD_ITEM = Symbol('add apd item');
export const EDIT_APD = Symbol('edit apd');
export const REMOVE_APD_ITEM = Symbol('remove apd item');

export const addKeyPerson = () => (dispatch, getState) => {
  dispatch({
    type: ADD_APD_ITEM,
    path: '/keyPersonnel/-',
    state: getState()
  });
};

export const removeKeyPerson = (
  index,
  { global = window } = {}
) => dispatch => {
  if (global.confirm('Do you really want to delete this key person?')) {
    dispatch({
      type: REMOVE_APD_ITEM,
      path: `/keyPersonnel/${index}`
    });
    dispatch(updateBudget());
  }
};

export const setKeyPersonName = (name, index) => ({
  type: EDIT_APD,
  path: `/keyPersonnel/${index}/name`,
  value: name
});

export const setKeyPersonEmail = (email, index) => ({
  type: EDIT_APD,
  path: `/keyPersonnel/${index}/email`,
  value: email
});

export const setKeyPersonRole = (role, index) => ({
  type: EDIT_APD,
  path: `/keyPersonnel/${index}/position`,
  value: role
});

export const setKeyPersonPercentTime = (percent, index) => ({
  type: EDIT_APD,
  path: `/keyPersonnel/${index}/percentTime`,
  value: percent
});

// needs to trigger budget calculation
export const setKeyPersonHasCosts = (hasCosts, index) => dispatch => {
  dispatch({
    type: EDIT_APD,
    path: `/keyPersonnel/${index}/hasCosts`,
    value: hasCosts
  });
  dispatch(updateBudget());
};

// needs to trigger budget calculation
export const setKeyPersonCost = (cost, index, year) => dispatch => {
  dispatch({
    type: EDIT_APD,
    path: `/keyPersonnel/${index}/costs/${year}`,
    value: cost
  });
  dispatch(updateBudget());
};

export const setMedicaidDirectorEmail = email => {
  return {
    type: EDIT_APD,
    path: '/stateProfile/medicaidDirector/email',
    value: email
  };
};

export const setMedicaidDirectorName = name => {
  return {
    type: EDIT_APD,
    path: '/stateProfile/medicaidDirector/name',
    value: name
  };
};

export const setMedicaidDirectorPhoneNumber = phone => {
  return {
    type: EDIT_APD,
    path: '/stateProfile/medicaidDirector/phone',
    value: phone
  };
};

export const setMedicaidOfficeAddress1 = address => {
  return {
    type: EDIT_APD,
    path: '/stateProfile/medicaidOffice/address1',
    value: address
  };
};

export const setMedicaidOfficeAddress2 = address => {
  return {
    type: EDIT_APD,
    path: '/stateProfile/medicaidOffice/address2',
    value: address
  };
};

export const setMedicaidOfficeCity = city => {
  return {
    type: EDIT_APD,
    path: '/stateProfile/medicaidOffice/city',
    value: city
  };
};

export const setMedicaidOfficeState = state => {
  return {
    type: EDIT_APD,
    path: '/stateProfile/medicaidOffice/state',
    value: state
  };
};

export const setMedicaidOfficeZip = zip => {
  return {
    type: EDIT_APD,
    path: '/stateProfile/medicaidOffice/zip',
    value: zip
  };
};
