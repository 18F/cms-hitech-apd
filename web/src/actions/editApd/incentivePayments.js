import { EDIT_APD } from './symbols';

export const setIncentiveEHCount = (year, quarter, value) => ({
  type: EDIT_APD,
  path: `/incentivePayments/ehCt/${year}/${quarter}`,
  value
});

export const setIncentiveEHPayment = (year, quarter, value) => ({
  type: EDIT_APD,
  path: `/incentivePayments/ehAmt/${year}/${quarter}`,
  value
});

export const setIncentiveEPCount = (year, quarter, value) => ({
  type: EDIT_APD,
  path: `/incentivePayments/epCt/${year}/${quarter}`,
  value
});

export const setIncentiveEPPayment = (year, quarter, value) => ({
  type: EDIT_APD,
  path: `/incentivePayments/epAmt/${year}/${quarter}`,
  value
});
