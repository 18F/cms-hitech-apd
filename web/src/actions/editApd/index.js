export * from './symbols';

export * from './keyPersonnel';
export * from './overview';
export * from './previousActivities';
export * from './stateProfile';
