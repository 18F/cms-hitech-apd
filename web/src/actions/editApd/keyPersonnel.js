import { ADD_APD_ITEM, EDIT_APD, REMOVE_APD_ITEM } from './symbols';
import { updateBudget } from '../apd';

export const addKeyPerson = () => (dispatch, getState) => {
  dispatch({
    type: ADD_APD_ITEM,
    path: '/keyPersonnel/-',
    state: getState()
  });
};

export const removeKeyPerson = (
  index,
  { global = window } = {}
) => dispatch => {
  if (global.confirm('Do you really want to delete this key person?')) {
    dispatch({
      type: REMOVE_APD_ITEM,
      path: `/keyPersonnel/${index}`
    });
    dispatch(updateBudget());
  }
};

export const setKeyPersonName = (name, index) => ({
  type: EDIT_APD,
  path: `/keyPersonnel/${index}/name`,
  value: name
});

export const setKeyPersonEmail = (email, index) => ({
  type: EDIT_APD,
  path: `/keyPersonnel/${index}/email`,
  value: email
});

export const setKeyPersonRole = (role, index) => ({
  type: EDIT_APD,
  path: `/keyPersonnel/${index}/position`,
  value: role
});

export const setKeyPersonPercentTime = (percent, index) => ({
  type: EDIT_APD,
  path: `/keyPersonnel/${index}/percentTime`,
  value: percent
});

// needs to trigger budget calculation
export const setKeyPersonHasCosts = (hasCosts, index) => dispatch => {
  dispatch({
    type: EDIT_APD,
    path: `/keyPersonnel/${index}/hasCosts`,
    value: hasCosts
  });
  dispatch(updateBudget());
};

// needs to trigger budget calculation
export const setKeyPersonCost = (cost, index, year) => dispatch => {
  dispatch({
    type: EDIT_APD,
    path: `/keyPersonnel/${index}/costs/${year}`,
    value: cost
  });
  dispatch(updateBudget());
};
