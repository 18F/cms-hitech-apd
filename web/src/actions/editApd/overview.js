import { ADD_APD_YEAR, EDIT_APD, REMOVE_APD_YEAR } from './symbols';
import { updateBudget } from '../apd';

export const addYear = year => (dispatch, getState) => {
  dispatch({ type: ADD_APD_YEAR, value: year, state: getState() });
  dispatch(updateBudget());
};

export const removeYear = year => (dispatch, getState) => {
  dispatch({ type: REMOVE_APD_YEAR, value: year, state: getState() });
  dispatch(updateBudget());
};

export const setNarrativeForHIE = text => ({
  type: EDIT_APD,
  path: '/narrativeHIE',
  value: text
});

export const setNarrativeForHIT = text => ({
  type: EDIT_APD,
  path: '/narrativeHIT',
  value: text
});

export const setNarrativeForMMIS = text => ({
  type: EDIT_APD,
  path: '/narrativeMMIS',
  value: text
});

export const setProgramOverview = text => ({
  type: EDIT_APD,
  path: '/programOverview',
  value: text
});
