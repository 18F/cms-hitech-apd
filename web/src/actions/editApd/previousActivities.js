import { EDIT_APD } from './symbols';

export const setPreviousActivitySummary = summary => ({
  type: EDIT_APD,
  path: '/previousActivitySummary',
  value: summary
});

export const setPreviousActivityApprovedExpenseForHITandHIE = (
  year,
  expense
) => ({
  type: EDIT_APD,
  path: `/previousActivityExpenses/${year}/hithie/totalApproved`,
  value: expense
});

export const setPreviousActivityFederalActualExpenseForHITandHIE = (
  year,
  expense
) => ({
  type: EDIT_APD,
  path: `/previousActivityExpenses/${year}/hithie/federalActual`,
  value: expense
});

export const setPreviousActivityApprovedExpenseforMMIS50FFP = (
  year,
  expense
) => ({
  type: EDIT_APD,
  path: `/previousActivityExpenses/${year}/mmis/50/totalApproved`,
  value: expense
});

export const setPreviousActivityApprovedExpenseforMMIS75FFP = (
  year,
  expense
) => ({
  type: EDIT_APD,
  path: `/previousActivityExpenses/${year}/mmis/75/totalApproved`,
  value: expense
});

export const setPreviousActivityApprovedExpenseforMMIS90FFP = (
  year,
  expense
) => ({
  type: EDIT_APD,
  path: `/previousActivityExpenses/${year}/mmis/90/totalApproved`,
  value: expense
});

export const setPreviousActivityFederalActualExpenseforMMIS50FFP = (
  year,
  expense
) => ({
  type: EDIT_APD,
  path: `/previousActivityExpenses/${year}/mmis/50/federalActual`,
  value: expense
});

export const setPreviousActivityFederalActualExpenseforMMIS75FFP = (
  year,
  expense
) => ({
  type: EDIT_APD,
  path: `/previousActivityExpenses/${year}/mmis/75/federalActual`,
  value: expense
});

export const setPreviousActivityFederalActualExpenseforMMIS90FFP = (
  year,
  expense
) => ({
  type: EDIT_APD,
  path: `/previousActivityExpenses/${year}/mmis/90/federalActual`,
  value: expense
});
