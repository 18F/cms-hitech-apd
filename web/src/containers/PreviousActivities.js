import React from 'react';
import { useDispatch, useSelector } from 'react-redux';

import { setPreviousActivitySummary } from '../actions/editApd';
import RichText from '../components/RichText';
import { Section, Subsection } from '../components/Section';
import ApdPreviousActivityTableHI from './ApdPreviousActivityTable';
import ApdPreviousActivityTableMMIS from './ApdPreviousActivityTableMMIS';
import ApdPreviousActivityTableTotal from './ApdPreviousActivityTableTotal';
import Waypoint from './ConnectedWaypoint';
import { t } from '../i18n';

const PreviousActivities = () => {
  const dispatch = useDispatch();

  const onChange = value => dispatch(setPreviousActivitySummary(value));

  const previousActivitySummary = useSelector(
    state => state.apd.data.previousActivitySummary
  );

  return (
    <Waypoint id="prev-activities-overview">
      <Section isNumbered id="prev-activities" resource="previousActivities">
        <Waypoint id="prev-activities-outline" />
        <Subsection
          id="prev-activities-outline"
          resource="previousActivities.outline"
        >
          <h4 className="ds-c-label">
            {t('previousActivities.outline.instruction.label')}
          </h4>
          <RichText
            content={previousActivitySummary}
            onSync={onChange}
            editorClassName="rte-textarea-l"
          />
        </Subsection>

        <Waypoint id="prev-activities-table" />
        <Subsection
          id="prev-activities-table"
          resource="previousActivities.actualExpenses"
        >
          <ApdPreviousActivityTableHI />
          <ApdPreviousActivityTableMMIS />
          <ApdPreviousActivityTableTotal />
        </Subsection>
      </Section>
    </Waypoint>
  );
};

export default PreviousActivities;
