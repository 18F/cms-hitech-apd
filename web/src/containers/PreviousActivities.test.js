import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import React from 'react';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import PreviousActivities from './PreviousActivities';
import { EDIT_APD } from '../actions/editApd';

// The rich text editor explodes in Enzyme - literally it exhausts v8's memory
// and crashes. So we can mock it to a do-nothing, since we're not actually
// interested in testing it here.
jest.mock('../components/RichText');

// Also mock out these tables, because we don't want to test them here.
jest.mock('./ApdPreviousActivityTable');
jest.mock('./ApdPreviousActivityTableMMIS');
jest.mock('./ApdPreviousActivityTableTotal');

const mockStore = configureStore([thunk]);

describe('previous activities component', () => {
  const store = mockStore({
    apd: { data: { previousActivitySummary: 'bob' } }
  });

  beforeEach(() => {
    store.clearActions();
  });

  test('renders correctly if data is not loaded', () => {
    const component = mount(
      <Provider store={store}>
        <PreviousActivities />
      </Provider>
    );
    expect(component).toMatchSnapshot();
  });

  test('updates on text change', () => {
    const component = mount(
      <Provider store={store}>
        <PreviousActivities />
      </Provider>
    );
    component.find('RichText').prop('onSync')('this is html');

    expect(store.getActions()).toEqual([
      {
        type: EDIT_APD,
        path: '/previousActivitySummary',
        value: 'this is html'
      }
    ]);
  });
});
