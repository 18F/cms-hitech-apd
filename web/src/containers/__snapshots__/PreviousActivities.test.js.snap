// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`previous activities component renders correctly if data is not loaded 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <PreviousActivities>
    <Connect(ConnectedWaypoint)
      id="prev-activities-overview"
    >
      <ConnectedWaypoint
        id="prev-activities-overview"
        scrollTo={[Function]}
      >
        <Waypoint
          bottomOffset="90%"
          debug={false}
          fireOnRapidScroll={true}
          horizontal={false}
          onEnter={[Function]}
          onLeave={[Function]}
          onPositionChange={[Function]}
          topOffset="0px"
        >
          <span
            style={
              Object {
                "fontSize": 0,
              }
            }
          />
        </Waypoint>
        <Section
          id="prev-activities"
          isNumbered={true}
          resource="previousActivities"
        >
          <section
            className="numbered-section"
            id="prev-activities"
          >
            <h2
              className="ds-h2"
            >
              Results of Previous Activities
            </h2>
            <span
              className="ds-text--lead"
            >
              Describe the results of the programâ€™s activities approved in the previous APD. This includes both the current status of those activities and the costs associated with them.
            </span>
            <Connect(ConnectedWaypoint)
              id="prev-activities-outline"
            >
              <ConnectedWaypoint
                id="prev-activities-outline"
                scrollTo={[Function]}
              >
                <Waypoint
                  bottomOffset="90%"
                  debug={false}
                  fireOnRapidScroll={true}
                  horizontal={false}
                  onEnter={[Function]}
                  onLeave={[Function]}
                  onPositionChange={[Function]}
                  topOffset="0px"
                >
                  <span
                    style={
                      Object {
                        "fontSize": 0,
                      }
                    }
                  />
                </Waypoint>
              </ConnectedWaypoint>
            </Connect(ConnectedWaypoint)>
            <Subsection
              headerClassName={null}
              id="prev-activities-outline"
              nested={false}
              resource="previousActivities.outline"
            >
              <h3
                className=" subsection--title ds-h3"
                id="prev-activities-outline"
              >
                Prior Activities Overview
              </h3>
              <Instruction
                args={null}
                headingDisplay={
                  Object {
                    "className": "ds-h3",
                    "level": "h3",
                  }
                }
                reverse={false}
                source="previousActivities.outline.instruction"
              >
                <div>
                  <div
                    className="visibility--screen"
                  >
                    <p
                      className="ds-u-font-weight--bold"
                    >
                      Summarize previous activities.
                    </p>
                    <Md
                      content="Provide a high level description of the activities that are completed, in-progress, or on-hold."
                      wrapper="p"
                    >
                      <p
                        dangerouslySetInnerHTML={
                          Object {
                            "__html": "Provide a high level description of the activities that are completed, in-progress, or on-hold.",
                          }
                        }
                      />
                    </Md>
                  </div>
                </div>
              </Instruction>
              <h4
                className="ds-c-label"
              >
                Previous Activities Summary
              </h4>
              <RichText
                content="bob"
                editorClassName="rte-textarea-l"
                onSync={[Function]}
              />
            </Subsection>
            <Connect(ConnectedWaypoint)
              id="prev-activities-table"
            >
              <ConnectedWaypoint
                id="prev-activities-table"
                scrollTo={[Function]}
              >
                <Waypoint
                  bottomOffset="90%"
                  debug={false}
                  fireOnRapidScroll={true}
                  horizontal={false}
                  onEnter={[Function]}
                  onLeave={[Function]}
                  onPositionChange={[Function]}
                  topOffset="0px"
                >
                  <span
                    style={
                      Object {
                        "fontSize": 0,
                      }
                    }
                  />
                </Waypoint>
              </ConnectedWaypoint>
            </Connect(ConnectedWaypoint)>
            <Subsection
              headerClassName={null}
              id="prev-activities-table"
              nested={false}
              resource="previousActivities.actualExpenses"
            >
              <h3
                className=" subsection--title ds-h3"
                id="prev-activities-table"
              >
                Actual Costs
              </h3>
              <Instruction
                args={null}
                headingDisplay={
                  Object {
                    "className": "ds-h3",
                    "level": "h3",
                  }
                }
                reverse={false}
                source="previousActivities.actualExpenses.instruction"
              >
                <div>
                  <div
                    className="visibility--screen"
                  >
                    <Md
                      content="Input the most recent approved and actual costs."
                      wrapper="p"
                    >
                      <p
                        dangerouslySetInnerHTML={
                          Object {
                            "__html": "Input the most recent approved and actual costs.",
                          }
                        }
                      />
                    </Md>
                  </div>
                </div>
              </Instruction>
              <Connect(ApdPreviousActivityTable) />
              <Connect(ApdPreviousActivityTableMMIS) />
              <Connect(ApdPreviousActivityTableMMIS) />
            </Subsection>
          </section>
        </Section>
      </ConnectedWaypoint>
    </Connect(ConnectedWaypoint)>
  </PreviousActivities>
</Provider>
`;
